manager/integration/tests/test_snapshot.py

```diff
@@ -0,0 +1,491 @@
+
+
+def test_snapshot_hash_global_disabled_with_immediate_hash(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_DISABLED, "true", "true")
+    check_per_volume_hash_disable(client, volume_name,
+                                  SNAPSHOT_DATA_INTEGRITY_IGNORED)
+
+
+def test_snapshot_hash_global_disabled_without_immediate_hash(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_DISABLED, "false", "true")
+    check_per_volume_hash_disable(client, volume_name,
+                                  SNAPSHOT_DATA_INTEGRITY_IGNORED)
+
+
+def test_snapshot_hash_global_disabled_and_per_volume_enabled_and_with_immediate_hash(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_DISABLED, "true", "true")
+    check_hash_calculated_and_with_immediate_hash(client, volume_name,
+                                                  SNAPSHOT_DATA_INTEGRITY_ENABLED)  # NOQA
+
+
+def test_snapshot_hash_global_disabled_and_per_volume_enabled_and_without_immediate_hash(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_DISABLED, "false", "true")
+    check_hash_calculated_and_without_immediate_hash(client, volume_name,
+                                                     SNAPSHOT_DATA_INTEGRITY_ENABLED)  # NOQA
+
+
+def test_snapshot_hash_global_enabled_and_per_volume_disable_and_with_immediate_hash(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_ENABLED, "true", "true")
+    check_per_volume_hash_disable(client, volume_name,
+                                  SNAPSHOT_DATA_INTEGRITY_DISABLED)
+
+
+def test_snapshot_hash_global_enabled_and_per_volume_enable_mode_without_immediate_hash(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_ENABLED, "false", "true")
+    check_per_volume_hash_disable(client, volume_name,
+                                  SNAPSHOT_DATA_INTEGRITY_DISABLED)
+
+
+def test_snapshot_hash_detect_corruption_in_global_enabled_mode(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_ENABLED, "true", "false")
+    detect_and_repair_corrupted_replica(client, volume_name,
+                                        SNAPSHOT_DATA_INTEGRITY_ENABLED)
+
+
+def test_snapshot_hash_detect_corruption_in_global_fast_check_mode(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_FAST_CHECK, "true",
+                    "false")
+    detect_and_repair_corrupted_replica(client, volume_name,
+                                        SNAPSHOT_DATA_INTEGRITY_FAST_CHECK)
+
+
+def update_settings(client, data_integrity, immediate_check, fast_replica_rebuild, period_in_second=SNAPSHOT_CHECK_PERIOD):  # NOQA
```

Comment:
Suggest renaming this to something more descriptive and less general, ex: `prepare_setting_for_snapshot_test`.

Reply:
Updated. Thank you.